src/domain/UserMoviesStorage/UserMoviesStorage.ts

```diff
@@ -0,0 +1,47 @@
+import { STORAGE_KEY } from '../../constants/INFORMATION';
+import DetailMovieData from '../../interfaces/DetailMovieData';
+
+const UserMoviesStorage = {
+  setUserMovies(userMovies: DetailMovieData[]): void {
+    localStorage.setItem(STORAGE_KEY, JSON.stringify(userMovies));
+  },
+
+  getUserMovies(): DetailMovieData[] {
+    const userMoviesInStorage = localStorage.getItem(STORAGE_KEY);
+
+    if (userMoviesInStorage != null) return JSON.parse(userMoviesInStorage);
+    return Array(0);
+  },
+
+  addUserMovie(userMovie: DetailMovieData) {
+    const userMoviesLocalStorage = this.getUserMovies();
+
+    userMoviesLocalStorage.push(userMovie);
+
+    this.setUserMovies(userMoviesLocalStorage);
```

Comment:
객체의 불변성을 위해서는 이렇게 해도 좋을 것 같아요
```suggestion
    this.setUserMovies([...userMoviesLocalStorage, userMovie]);
```

Reply:
스프레드 연산자에 대해서 공부는 했지만, 미션을 진행하면서 거의 사용을 못했었습니다. 아니 사용할 생각을 못했었네요! 그런데 이렇게 피드백을 만나보니 앞으로 구현할 때 생각 확장해볼 수 있을 것 같아서 좋습니다.

Reply:
맞습니다! 저는 기본적으로 객체를 변형시킬 수 있는 매서드들(push, pop, splice 등등)을 되도록이면 안 쓸 수 있는 방향으로 개발을 하는 것을 선호하는 것 같아요! 분명 나중에는 더 많은 데이터/객체를 한 번에 다루게 될 텐데, 이럴 때 예상하지 못한 사이드이펙트가 발생하는 것을 최소화할 수 있다고 생각합니다!